In [1]:
import urllib.request 
url = ("https://raw.githubusercontent.com/rasbt/" 
 "LLMs-from-scratch/main/ch02/01_main-chapter-code/" 
 "the-verdict.txt") 
file_path = "the-verdict.txt" 
urllib.request.urlretrieve(url, file_path)

('the-verdict.txt', <http.client.HTTPMessage at 0x22440c88050>)

通过 Python 读取短篇小说 The Verdict 作为文本样本

In [ ]:
with open("the-verdict.txt", "r", encoding="utf-8") as f: 
 raw_text = f.read() 
print("Total number of character:", len(raw_text)) 
print(raw_text[:99])



Total number of character: 20479
I HAD always thought Jack Gisburn rather a cheap genius--though a good fellow enough--so it was no 


打印（print）命令首先输出该文件的字符总数，然后展示文件的前 100 个字符作为内容示例：
Total number of character: 20479 
I HAD always thought Jack Gisburn rather a cheap genius--though a good fellow 
 enough--so it was no
我们的目标是将这篇包含 20 479 个字符的短篇小说分割为独立的单词和特殊字符，以便在后续
章节中将它们转换为嵌入向量，进而用于大语言模型训练。

In [1]:
import re 
text = "Hello, world. This, is a test." 
result = re.split(r'(\s)', text) 
print(result)

['Hello,', ' ', 'world.', ' ', 'This,', ' ', 'is', ' ', 'a', ' ', 'test.']


运行代码得到的结果是一个包含单个单词、空白字符和标点符号的列表：
['Hello,', ' ', 'world.', ' ', 'This,', ' ', 'is', ' ', 'a', ' ', 'test.']

In [2]:
result = [item for item in result if item.strip()]

print(result)

['Hello,', 'world.', 'This,', 'is', 'a', 'test.']
